pkg/curatedpackages/bundle.go

```diff
@@ -0,0 +1,100 @@
+package curatedpackages
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-logr/logr"
+
+	api "github.com/aws/eks-anywhere-packages/api/v1alpha1"
+	"github.com/aws/eks-anywhere-packages/pkg/artifacts"
+	"github.com/aws/eks-anywhere-packages/pkg/bundle"
+	"github.com/aws/eks-anywhere-packages/pkg/testutil"
+	"github.com/aws/eks-anywhere/pkg/constants"
+	"github.com/aws/eks-anywhere/pkg/dependencies"
+	"github.com/aws/eks-anywhere/pkg/executables"
+)
+
+const (
+	RegistryBaseRef = "public.ecr.aws/q0f6t3x4/eksa-package-bundles"
+)
+
+func GetLatestBundle(ctx context.Context, kubeConfig string, source BundleSource, kubeVersion string) (*api.PackageBundle, error) {
+	switch source {
```

Comment:
Instead of having a switch, you could make `BundleSource` and interface and make it implement `GetLatestBundle`